src/packages/shared-types/actions.ts

```diff
@@ -1,8 +1,37 @@
+import { OsMainSourceItem } from "./opensearch";
+import { CognitoUserAttributes } from "./user";
+import { LatestRai } from "shared-utils";
+import { SEATOOL_STATUS } from "./statusHelper";
+
 export enum Action {
+  ISSUE_RAI = "issue-rai",
+  RESPOND_TO_RAI = "respond-to-rai",
   ENABLE_RAI_WITHDRAW = "enable-rai-withdraw",
-  WITHDRAW_PACKAGE = "withdraw-package",
   DISABLE_RAI_WITHDRAW = "disable-rai-withdraw",
-  ISSUE_RAI = "issue-rai",
   WITHDRAW_RAI = "withdraw-rai",
-  RESPOND_TO_RAI = "respond-to-rai",
+  WITHDRAW_PACKAGE = "withdraw-package",
 }
+const raiStatuses = [
+  SEATOOL_STATUS.PENDING,
+  SEATOOL_STATUS.PENDING_OFF_THE_CLOCK,
+  SEATOOL_STATUS.PENDING_APPROVAL,
+  SEATOOL_STATUS.PENDING_CONCURRENCE,
+];
+export const ActionAvailabilityCheck = {
+  isInRaiStatus: (seatoolStatus: string) => raiStatuses.includes(seatoolStatus),
+  isNotWithdrawn: (seatoolStatus: string) =>
```

Comment:
Something to chew on: which of the following implementations would be preferable and why?

1. 
```typescript
ActionAvailabilityCheck(data?.seatoolStatus).isInRaiStatus // direct boolean
```

2.
```typescript
ActionAvailabilityCheck.isInRaiStatus(data?.seatoolStatus) // called method, returns boolean
```

Reply:
1️⃣

Reply:
I like the idea of having methods like `is` and `can`, but the idea is to balance use of the "catch-all" method like `AuthorityService.is()` with our most common checks or more complex checks so method params can be baked into them and tested while bugs are relegated to the ad-hoc uses of the catch-all.

Like, the array `[ SEATOOL_STATUS.PENDING, SEATOOL_STATUS.PENDING_OFF_THE_CLOCK, SEATOOL_STATUS.PENDING_APPROVAL]` is basically second clock statuses, and when paired with other intrinsic attributes and conditions, you can use something like `PackageService(data).isInSecondClock` as a boolean check. 

One thing I have thought about, too, is chaining checks on a single object type, akin to `PackageService(data).isInSecondClock().canWithdrawRaiResponse()` but without needing to return the object from a method each time. That's not an immediate need as we can just do `checks.isInSecondClock && checks.canWithdrawRai` but compound conditions like that also get remade a lot without need for ad-hoc control. I know a common pattern is `PackageService(data).all([...])` where you can give it some value and it'll run all checks.